### ISSUE TITLE
difficulty with svg

### ISSUE THREAD

**Issue by JerrySievert** [VCVRack/Tutorial#8]
hi, sorry to post this as an issue - I'm hoping it's something that can be a quick response (and if I'm not the only one being an idiot when it comes to this, perhaps added to some documentation).

I've gone through the tutorial, and have a working module that does exactly what I want it to do, but the svg that I'm generating seems to be correct, but when compiled the panel is transparent.

export looks to be correct (I'm using photoshop and exporting as svg, and tried Inkscape initially with the same results), and renders correctly in a web browser.

I'm happy to post the svg somewhere - any advice would be awesome.

thanks!

**Comment by AndrewBelt** [comment]
Does `log.txt` say that you're loading the correct file path?

**Comment by djpeterso23662** [comment]
Hi Jerry!  I am guessing your issues are generated from including Photoshop in your workflow.  When creating your SVG, bear in mind that VCV Rack supports two basic elements in Inkscape SVG--  paths and solid fills.  So you have to work within those constraints.  If you experiment by starting with an Inkscape SVG with a rectangle with a solid fill, and work up from there, you should get to the finish line quickly.  An SVG with an imported bitmap will not show the bitmap.

**Comment by JerrySievert** [comment]
@AndrewBelt where would I find `log.txt`? the compilation was error and warning free.

**Comment by JerrySievert** [comment]
@djpeterso23662 that might be the case - I'll try to get more proficient with Inkscape (I'm not a big fan of it, tbh), and will check to figure out if I can make even a rectangle with a solid fill will work (the one that I tried from Inkscape was also transparent in Rack).

great work on Rack, btw, I'm using it to prototype hardware I'm designing for my physical eurorack :)

**Comment by djpeterso23662** [comment]
@JerrySievert you can save a little time by just using the Tutorial SVG as-is.  If it shows up you know everything else is correct.  

I have also heard of people having transparent modules when the SVG was not in the right place.  That is, in the res folder within the module folder, if your setup closely follows Tutorial.

**Comment by JerrySievert** [comment]
@djpeterso23662 yes, the tutorial svg is loading correctly when I overwrite the svg I created:

`cp Tutorial/res/MyModule.svg SynthKit/res/Addition.svg`

so, that at least seems to be correct.  the annoying this is that I didn't rasterize anything in photoshop, so it should have stayed vector as paths.

when I finally get this figured out, I'll try to put in a PR with some additional svg tips.

**Comment by AndrewBelt** [comment]
I don't accept PRs, but you're welcome to use the issue tracker for any type of discussions.

**Comment by JerrySievert** [comment]
just curious (I maintain lots of open source projects) as to why you don't accept PRs.

**Comment by djpeterso23662** [comment]
@JerrySievert yes, thank you very much, SVG tips would be a very welcome addition.  As Andrew said already, this issue tracker is a good place to post.  I can share my Inkscape tips, humble as they are, as well.  A short list might be:

Edit the Tutorial.svg file, resizing the background layer to your dimensions.  Set it to a solid RGB fill you are happy with.  

Use the font-to-path menu to convert type to paths.  

Include only lines, paths, and shapes with solid or no fills.  Inkscape's feature to crop one shape with another is not supported, so a box matching the background color is needed to do that stuff.  

Good luck!

**Comment by JerrySievert** [comment]
I moved from using Inkscape to generating my svg's using `svg.js`.  it might be worth adding a note that, as well as a quick example of how to build it.

**Comment by squinkylabs** [comment]
@JerrySievert If VCVRack doesn't want to take PRs, maybe "the community" needs to set up a source for documentation and tips on writing modules. Maybe you could admin a repo just for third party documentation!

**Comment by AndrewBelt** [comment]
@squinkylabs:
@jeremywen maintains this wiki, which is publically editable.
https://github.com/jeremywen/JW-Modules/wiki/Dev-FAQ-for-VCV-Rack

**Comment by JerrySievert** [comment]
Beyond the wiki, I'm also writing a blog entry aimed at developers (and have been helping some friends on their module writing quests) so there will be more docs regardless.

**Comment by squinkylabs** [comment]
@JerrySievert is the blog up now?

**Comment by JerrySievert** [comment]
The blog itself is but not the vcvrack entry.
